spec/fixtures/custom_plugins/kong/plugins/reconfiguration-completion/handler.lua

```diff
@@ -0,0 +1,32 @@
+local kong_meta = require "kong.meta"
+
+local ReconfigurationCompletionHandler = {
+  VERSION = kong_meta.version,
+  PRIORITY = 2000000,
+}
+
+
+function ReconfigurationCompletionHandler:rewrite(conf)
+  kong.response.set_header("X-Kong-Reconfiguration-Status", "unknown")
+end
+
+
+function ReconfigurationCompletionHandler:access(conf)
+  local if_kong_configuration_version = kong.request and kong.request.get_header('if-kong-configuration-version')
+  if if_kong_configuration_version then
+    if if_kong_configuration_version ~= conf.version then
+      return kong.response.error(
+        503,
+        "Service Unavailable",
+        {
+          ["X-Kong-Reconfiguration-Status"] = "pending",
+          ["Retry-After"] = tostring((kong.configuration.worker_state_update_frequency or 1) + 1),
+        }
+      )
+    else
+      kong.response.set_header("X-Kong-Reconfiguration-Status", "complete")
+    end
+  end
+end
```

Comment:
Why not run all this on rewrite? Perhaps you could even short-circuit the request there? This of course means it needs to be global, but I think this plugin can be global only.

Reply:
I've consolidated everything into the `rewrite` function.  The plugin is only used globally anyway.